alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/transport/TransportIndexMonitorAction.kt

```diff
@@ -379,24 +379,31 @@ class TransportIndexMonitorAction @Inject constructor(
             val getRequest = GetRequest(SCHEDULED_JOBS_INDEX, request.monitorId)
             client.get(getRequest, object : ActionListener<GetResponse> {
                 override fun onResponse(response: GetResponse) {
-                    onGetResponse(response)
+                    if (!response.isExists) {
+                        actionListener.onFailure(AlertingException.wrap(
+                            ElasticsearchStatusException("Monitor with ${request.monitorId} is not found", RestStatus.NOT_FOUND)))
+                        return
+                    }
+                    val xcp = XContentHelper.createParser(xContentRegistry, LoggingDeprecationHandler.INSTANCE,
+                        response.sourceAsBytesRef, XContentType.JSON)
+                    val monitor = ScheduledJob.parse(xcp, response.id, response.version) as Monitor
+                    onGetResponse(monitor)
                 }
                 override fun onFailure(t: Exception) {
                     actionListener.onFailure(AlertingException.wrap(t))
                 }
             })
         }
 
-        private fun onGetResponse(response: GetResponse) {
-            if (!response.isExists) {
+        private fun onGetResponse(currentMonitor: Monitor) {
+            val backendRoles = currentMonitor.user?.backendRoles
+            if (filterByEnabled &&
+                (backendRoles == null || user?.backendRoles == null || backendRoles.intersect(user.backendRoles).isEmpty())) {
                 actionListener.onFailure(AlertingException.wrap(
-                        ElasticsearchStatusException("Monitor with ${request.monitorId} is not found", RestStatus.NOT_FOUND)))
-                return
+                    ElasticsearchStatusException("Do not have backend roles to delete monitor with ${request.monitorId}",
```

Comment:
Can we make the message more generic? `Do not have permissions to delete monitor with ...`

Reply:
Done